orchestrator/careplancontributor/service.go

```diff
@@ -172,43 +213,89 @@ func (s Service) createCarePlan(patient fhir.Patient) (*fhir.CarePlan, error) {
+
 	task := map[string]any{
 		"resourceType": "Task",
 		"status":       "requested",
 		"intent":       "order",
 		"requester":    serviceRequest.Requester,
 		"owner":        serviceRequest.Performer,
 		"reasonCode":   serviceRequest.Code,
-		// "for": fhir.Reference{
-		// 	Type:      to.Ptr("Patient"),
-		// 	Reference: patient.Id,
-		// },
-		// "input": []map[string]any{
-		// 	{
-		// 		"valueReference": fhir.Reference{
-		// 			Type:      to.Ptr("ServiceRequest"),
-		// 			Reference: serviceRequest.Id,
-		// 		},
-		// 	},
-		// },
-		// "contained": []interface{}{
-		// 	serviceRequest, patient,
-		// },
 	}
 	taskJSON, _ := json.MarshalIndent(task, "", "  ")
 	println(string(taskJSON))
 	createdTask, err := coolfhir.Workflow{CarePlanService: s.CarePlanService}.Invoke(task)
 	return createdTask, err
 }
 
+// When an application accepts the Task, we enrich the Task with more detailed information like the actual Patient and the ServiceRequest
+func (s Service) handleAcceptedTask(task *fhir.Task, serviceRequest *fhir.ServiceRequest, patient *fhir.Patient) (*fhir.Task, error) {
+
+	taskJSON, err := s.enrichTask(task, serviceRequest, patient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to enrich task: %w", err)
+	}
+
+	var enrichedTask fhir.Task
+	if err := s.CarePlanService.Update("Task/"+*task.Id, json.RawMessage(*taskJSON), &enrichedTask); err != nil {
+		return nil, fmt.Errorf("failed to update Task: %w", err)
+	}
+
+	// s.updateCarePlanWithTask(result)
+
+	//carePlan.Activity = append(carePlan.Activity, fhir.CarePlanActivity{
+	//	Reference: &fhir.Reference{
+	//		Type:      to.Ptr("Task"),
+	//		Reference: to.Ptr("Task/" + *task.Id), // TODO: This seems a fiddly way to reference stuff
+	//	},
+	//})
+	//// TODO: Add "If" headers to make sure we're not overwriting someone else's changes
+	//if err := s.CarePlanService.Update("CarePlan/"+*carePlan.Id, carePlan, &carePlan); err != nil {
+	//	return nil, fmt.Errorf("failed to add Task to CarePlan: %w", err)
+	//}
+
+	return &enrichedTask, nil
+}
+
+func (s Service) enrichTask(task *fhir.Task, serviceRequest *fhir.ServiceRequest, patient *fhir.Patient) (*[]byte, error) {
+
+	//FIXME: fhir.Task.Contained does not exist for some reason - converting the task to a map for easy manipulation instead
+	taskMap := make(map[string]interface{})
+	taskJSON, _ := json.Marshal(task)
+	if err := json.Unmarshal(taskJSON, &taskMap); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal task: %w", err)
+	}
+
+	taskMap["for"] = fhir.Reference{
+		Type:      to.Ptr("Patient"),
+		Reference: to.Ptr("#" + *patient.Id), //convert to local reference
+	}
+
+	taskMap["input"] = []map[string]interface{}{
+		{
+			"valueReference": fhir.Reference{
+				Type:      to.Ptr("ServiceRequest"),
+				Reference: to.Ptr("#" + *serviceRequest.Id), //convert to local reference
+			},
+		},
+	}
+
+	taskMap["contained"] = []interface{}{
+		*serviceRequest,
+		*patient,
+	}
+
+	taskJSON, _ = json.Marshal(taskMap)
```

Comment:
the FHIR library already does (un)marshalling, so no need to marshal here. Just return the `Task` map.